### PR TITLE
Add FXIOS-9269 Create and add error toast for address saving failure

### PR DESCRIPTION
**Pull request by razvanlitianu**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9269)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/20521)

## :bulb: Description

This PR introduces toast notifications to inform users of save address process failures.

Failure Notifications:
Immediate alert when the save/update operation fails.
Feedback for persistent failures after all retries.

Error then success
https://github.com/mozilla-mobile/firefox-ios/assets/26011662/9f452fc7-a18c-4592-8bfc-5d94380ab4a1

Retry failure
https://github.com/mozilla-mobile/firefox-ios/assets/26011662/140347d5-9d00-4c1e-b6b1-248f6ff76794

## :pencil: Checklist
You have to check all boxes before merging
- [ ] Filled in the above information (tickets numbers and description of your work)
- [ ] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Client/Frontend/Autofill/Address/Edit/AddressModifiedStatus.swift [hunk 1]
```diff
@@ -5,15 +5,42 @@
...
+    enum ErrorType {
+        case save(action: () -> Void)
+        case update(action: () -> Void)
+        case remove(action: () -> Void)
+
+        var message: String {
+            switch self {
+            case .save: return .Addresses.Settings.Edit.AddressSaveError
+            case .update: return .Addresses.Settings.Edit.AddressSaveError
+            case .remove: return .Addresses.Settings.Edit.AddressRemoveError
```
Comment (nbhasin2) [inline firefox-ios/Client/Frontend/Autofill/Address/Edit/AddressModifiedStatus.swift:17]: Do they all have same error? 🤔

Comment (razvanlitianu) [reply firefox-ios/Client/Frontend/Autofill/Address/Edit/AddressModifiedStatus.swift:17]: Only the saving and updating.

**Review by nbhasin2 — APPROVED** [review]
@issammani double approval from you but lgtm

File: firefox-ios/Client/Frontend/Autofill/Address/Edit/AddressModifiedStatus.swift (hunk 1 above)
Comment (nbhasin2) [reply firefox-ios/Client/Frontend/Autofill/Address/Edit/AddressModifiedStatus.swift:17]: @razvanlitianu debating if we should combine this but R+ regardless 

```suggestion
            case .save, .update: return .Addresses.Settings.Edit.AddressSaveError
            case .remove: return .Addresses.Settings.Edit.AddressRemoveError
```

**Pull request merged**